content/blog/2017/07/2017-07-22-powershell-pipeline.adoc

```diff
@@ -0,0 +1,198 @@
+Another interesting aspect of this example is that the _powershell_ build step failed, which ultimately caused
+the job to fail. The failure in this example is due to the PowerShell error stream being non-empty, which
+therefore caused the step to result in a non-zero exit status. However,
+as you will soon discover, there are a variety of causes for a failing _powershell_ build step.
+
+=== What causes a failing exit status?
+
+When you execute a _powershell_ build step, it may produce a non-zero exit code and fail your pipeline build.
+This is very similar to other shell steps with some interesting caveats. Your _powershell_ step 
+may produce a failing exit status in the following instances:
+
+1. Something in your PowerShell script has thrown an exception
+2. Your PowerShell script explicitly calls _exit_ with a non-zero exit code
+3. Your PowerShell script calls a native application that produces a non-zero $LastExitCode
+* $LastExitCode is an link:https://msdn.microsoft.com/en-us/powershell/reference/5.1/microsoft.powershell.core/about/about_automatic_variables[automatic variable] that is set after executing a native application
+4. Your PowerShell script results in a non-empty error stream (with or without throwing an exception)
+
+Overriding the exit status behavior of your _powershell_ build step can be achieved by explicitly exiting
+from your script as long as the failure was not caused by an unhandled exception. For example:
+
+==== Unavoidable failure caused by an unhandled exception
+
+[source,groovy]
+----
+node {
+    powershell '''
+        throw 'Error! Problem Exists Between Keyboard And Chair'
+        exit 0  # Unreachable code
+    '''
+}
+----
+
+==== Failed step caused by a non-empty error stream
+
+[source,groovy]
+----
+node {
+    powershell '''
+        Write-Error 'Error! Problem Exists Between Keyboard And Chair'
+    '''
+}
+----
+
+==== Failure prevented by an explicit exit
+
+[source,groovy]
+----
+node {
+    powershell '''
+        Write-Error 'Error! Problem Exists Between Keyboard And Chair'
+        exit 0
+    '''
+}
+----
+
+=== Scripts vs. Cmdlets
+
+A Cmdlet is a small lightweight utility written in either C#, and compiled, or written in PowerShell directly. Depending on what your goal is in your pipeline you can make
+use of Cmdlets directly in your pipeline code, call a self contained PowerShell script, or some mixture of the two. If your strategy is to keep each _powershell_ step as
+short and succinct as possible then it may make sense for you to write a library of Cmdlets, but if you have monolithic scripts then it may make sense for you to call those 
+scripts directly from your pipeline. The choice is entirely up to you, as both scenarios are supported.
+
+== Thanks for reading, and have fun!
+
+I sincerely hope that this post has encouraged you to try using PowerShell in your Jenkins pipeline. Please do not hesitate to file an issue on GitHub if you have discovered any
```

Comment:
It would be beneficial to link to the actual repository toward which users should file issues here.